### PR TITLE
MRG: Adjust lineno for 3.8

### LINKED ISSUE #442: gallery generation broken on cpython master
Have not had time to dig into why (or have a minimal example), but seeing things like

```
jupiter@11:39  ➤  make html
Running Sphinx v1.8.4
Adding copy buttons to code blocks...
[autosummary] generating autosummary for: api/_as_gen/matplotlib.animation.AVConvBase.rst, api/_as_gen/matplotlib.animation.AVConvFileWriter.rst, api/_as_gen/matplotlib.animation.AVConvWriter.rst, api/_as_gen/matplotlib.animation.AbstractMovieWriter.rst, api/_as_gen/matplotlib.animation.Animation.rst, api/_as_gen/matplotlib.animation.ArtistAnimation.rst, api/_as_gen/matplotlib.animation.FFMpegBase.rst, api/_as_gen/matplotlib.animation.FFMpegFileWriter.rst, api/_as_gen/matplotlib.animation.FFMpegWriter.rst, api/_as_gen/matplotlib.animation.FileMovieWriter.rst, ..., users/prev_whats_new/whats_new_1.2.2.rst, users/prev_whats_new/whats_new_1.2.rst, users/prev_whats_new/whats_new_1.3.rst, users/prev_whats_new/whats_new_1.4.rst, users/prev_whats_new/whats_new_1.5.rst, users/prev_whats_new/whats_new_2.0.0.rst, users/prev_whats_new/whats_new_2.1.0.rst, users/prev_whats_new/whats_new_2.2.rst, users/shell.rst, users/whats_new.rst
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://docs.scipy.org/doc/numpy/objects.inv...
loading intersphinx inventory from https://docs.scipy.org/doc/scipy/reference/objects.inv...
loading intersphinx inventory from https://pandas.pydata.org/pandas-docs/stable/objects.inv...
loading intersphinx inventory from https://pillow.readthedocs.io/en/stable/objects.inv...
loading intersphinx inventory from https://matplotlib.org/cycler/objects.inv...
generating gallery...
generating gallery for gallery/lines_bars_and_markers... [  2%] bar_stacked.py                                                                                                                                                                                                                       
Warning, treated as error:
/home/tcaswell/source/p/matplotlib/examples/lines_bars_and_markers/bar_stacked.py failed to execute correctly: Traceback (most recent call last):
  File "/home/tcaswell/source/p/matplotlib/examples/lines_bars_and_markers/bar_stacked.py", line 1
    =================
    ^
SyntaxError: invalid syntax

make: *** [Makefile:30: html] Error 2

```
when trying to build the Matplotlib docs.

This makes me think that the module docstring is not being correctly identified as such.  Similar things have happened before when details of the AST changed.

### PR DESCRIPTION
**Pull request by larsoner**
This gets `sphinx-gallery` working locally on 3.8.alpha2 for me.

Closes #442

### CODE REVIEW

**Comment by larsoner** [comment]
Ready for review/merge from my end. @tacaswell this touches the 3.7 lines you added a while back (the `.docstring` appears to have been reverted so I updated it to use `ast.get_docstring`)

File: setup.cfg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tool:pytest]
-addopts = --cov-report term-missing --cov=sphinx_gallery --durations=5 -rs
+addopts = --cov-report= --cov=sphinx_gallery --durations=5 -rs
```
Comment (choldgraf) [inline setup.cfg]: is the hanging `=` intentional? I'm not sure, just feels like it might be a big to me :-)

Comment (larsoner) [reply setup.cfg]: Yes it's intentional, it generates the file but prints nothing by default. It makes debugging (much more common than coverage checking) much easier, and can still be easily overridden when someone wants the long report

**Comment by choldgraf** [comment]
@larsoner could you clarify the merge conflict and then we can merge?

**Comment by tacaswell** [comment]
Thanks!

**Pull request merged**